lib/legacy/class.js

```diff
@@ -211,9 +215,89 @@ function GenerateClassFromInfo(info, Base, behaviors) {
   let behaviorList;
   const lifecycle = {};
 
+  // Work around for closure bug #126934458. Using `super` in a property
+  // getter does not work so instead we search the Base prototype for an
+  // implementation of observedAttributes so that we can override and call
+  // the `super` getter. Note, this is done one time ever because we assume
+  // that `Base` is always comes from `Polymer.LegacyElementMixn`.
+  if (!observedAttributesGetter) {
+    let ctor = Base;
+    while (ctor && !observedAttributesGetter) {
+      const desc = Object.getOwnPropertyDescriptor(ctor, 'observedAttributes');
+      if (desc) {
+        observedAttributesGetter = desc.get;
+      }
+      ctor = Object.getPrototypeOf(ctor.prototype).constructor;
+    }
+  }
+
   /** @private */
   class PolymerGenerated extends Base {
 
+    static get observedAttributes() {
+      return observedAttributesGetter.call(this).concat(DISABLED_ATTR);
+    }
+
+    // Prevent element from initializing properties when it's upgrade disabled.
+    /** @override */
+    _initializeProperties() {
+      // Enable disable-upgrade use when `legacyOptimizations` setting is on.
+      if (legacyOptimizations && this.hasAttribute(DISABLED_ATTR)) {
+        this.__isUpgradeDisabled = true;
+      } else {
+        super._initializeProperties();
+      }
+    }
+
+    // Prevent element from enabling properties when it's upgrade disabled.
+    /** @override */
+    _enableProperties() {
+      if (!this.__isUpgradeDisabled) {
```

Comment:
Normally we would never get here, since this is normally called from `connectedCallback` and that also has a gate. However `Templatizer` boots up trees of elements by forcing `_enableUpgrade` on them, even if they're not connected (since booting elements out of the tree has shown to be performance positive). Worth a comment.